File: 08-DictionariesMaps/solution_08.scala
object Solution {
    def main(args: Array[String]) = {

        // Take all input
        val inputs: List[String] = io.Source.stdin.getLines.toList
        val nEntries: Int = inputs.head.toInt
        
        def splitToMap(str: String): (String, String) = {
            // Return Name + Number splits
            val splits = str.split(" ")
            (splits(0) -> splits(1))
        }
        
        val phonebook = inputs.tail.slice(0, nEntries).foldLeft[Map[String,String]](Map.empty)((listAcc: Map[String,String], e: String) => listAcc + splitToMap(e))

        
        def find(entry: String): String = phonebook.get(entry) match {
            // Find & return any query names that match an entry 
            case Some(i) => entry + "=" + i
            case None => "Not found"
        }

        val queries = inputs.tail.drop(nEntries)
        val outputs = queries.map(find(_))
        println(outputs.mkString("\n"))
    }
}

File: 08-DictionariesMaps/solution_08.py
# Generate phonebook dict and return queries afterward
n_entries = int(input().strip())  # Number of phonebook definitions incoming


def create_phonebook(n_entry):
    phonebook = dict()    # Takes nEntries as user input('Ethan 9923')

    for _i in range(0, n_entry):
        entry = str(input())

        split_entry = entry.split()

        name = split_entry[0]
        number = split_entry[1]

        phonebook[name] = number

    return phonebook


def query_phonebook(phonebook):
    while True:  # Run without conditions, until break statement executes
        try:
            query = str(input())
            if query in phonebook:
                print(str(query) + "=" + str(phonebook[query]))
            else:
                print("Not found")
        except:
            break


phonebook = create_phonebook(n_entries)
query_phonebook(phonebook)
